azurerm/internal/services/media/media_content_key_policy_resource.go

```diff
@@ -458,7 +453,17 @@ func resourceMediaContentKeyPolicyRead(d *schema.ResourceData, meta interface{})
 	d.Set("resource_group_name", id.ResourceGroup)
 	d.Set("media_services_account_name", id.MediaserviceName)
 	d.Set("description", resp.Description)
-	d.Set("policy_option", flattenPolicyOptions(resp.Options))
+
+	if resp.Options != nil {
+		options, err := flattenPolicyOptions(resp.Options)
+		if err != nil {
+			return err
+		}
+
+		if err := d.Set("policy_option", options); err != nil {
+			return fmt.Errorf("setting `policy_option`: %+v", err)
+		}
+	}
```

Comment:
can we remove this outer `if` here, the `flattenPolicyOptions` function returns a `[]interface{}` if this is nil, to ensure we track the diff